src/Adapter/OrderReturn/Repository/OrderReturnRepository.php

```diff
@@ -86,4 +110,122 @@ public function update(OrderReturn $orderReturn): void
             OrderReturnException::class
         );
     }
+
+    /**
+     * @param OrderReturnId $orderReturnId
+     * @param OrderReturnDetailId $orderReturnDetailId
+     *
+     * @throws CoreException
+     * @throws DeleteOrderReturnProductException
+     * @throws OrderReturnException
+     */
+    public function deleteOrderReturnProduct(
+        OrderReturnId $orderReturnId,
+        OrderReturnDetailId $orderReturnDetailId
+    ): void {
+        $orderReturn = $this->get($orderReturnId);
+
+        if ((int) ($orderReturn->countProduct()) <= 1) {
```

Comment:
Yes exactly, you can't delete last product in order return, because then that order return won't make sense.

Added comment to clarify I could change to = 1, to be more clear.